shm/map.cpp

```diff
@@ -0,0 +1,21 @@
+#include "map.hpp"
+
+#include <algorithm>
+#include <random>
+
+Map::Map() {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(0, 255);
+    for (size_t i = 0; i < 10; ++i) {
+        Island temp(distrib(gen), distrib(gen));
+        if (std::none_of(Islands_.begin(), Islands_.end(), [temp](const Island& islnd) {
+                return islnd.getCoordinates() == temp.getCoordinates();
+            })) {
+            Islands_.push_back(temp);
+        } else {
+            Island temp2(distrib(gen), distrib(gen));
+            Islands_.push_back(temp2);
+        }
```

Comment:
When second time you will generate already taken numbers you will have two islands with the same coordinates.
While loop should be better.

Reply:
Yes I know, but when I implemented while loop it couldn't see the temp variable both in while condition and in next lines of code